### ISSUE TITLE
The ModelicaStandardLibrary and associated libraries seem to be unavailable for install via apt on Ubuntu/Docker

### ISSUE THREAD

**Issue by ramanunni** [OpenModelica/OpenModelica#9595]
Hello there,
Up till recently, when one wanted to install the MSL or associated libraries either within a docker image or on Ubuntu (20.04/22.04), one could install it using:
`apt-get install -y omc omlib-modelica-3.2.2` OR `apt-get install omlib-modelica-4.0.0 omlib-complex-4.0.0 omlib-modelicaservices-4.0.0`

At present, only the install call for omc works, which installs the minimal Openmodelica Compiler.

Has there been a complete change of direction and is there a new mode of installation that is preferred even for minimal installations? I ask that as I see that from 1.20, there seems to be a move towards automatic install of libraries as per the discussion here: https://github.com/OpenModelica/OpenModelica/issues/9583

It is rather confusing as currently, I can't find any direction on the new preferred method anywhere, neither on the [OpenModelicaDockerImages](https://github.com/OpenModelica/OpenModelicaDockerImages) repository, nor on the [Jupyter-Openmodelica](https://github.com/OpenModelica/jupyter-openmodelica) repository or any other official repo. The Jupyter-Openmodelica repository still point towards the method I have mentioned above in line 12 of the [Dockerfile](https://github.com/OpenModelica/jupyter-openmodelica/blob/master/Dockerfile).

Can someone share some insight into this and clarify what is going on for installation of MSLv3.2.x/4.0.0 in Ubuntu/Docker when using 20.04/22.04 and using the 1.18 and 1.19 versions of Openmodelica? Are we going to lose the ability to use apt-get install permanently, which was so convenient for the versions of omc till now, especially if we wanted a minimal build.

**Comment by sjoelund** [comment]
> @sjoelund can you please comment on that? Did we remove the old omlib-modelica-x.x.x packages from the APT server?

Yes. There are even conflicts marked on the old packages so if you find them you can't install them.

> Although we strongly recommend people to always use the latest released version, we should of course allow people to use previous versions like 1.19.0 or 1.18.0.

These are still available at https://build.openmodelica.org/omc/builds/linux/releases/ including libraries

> This mechanism already works well in Windows. We just need to add the cached libraries to the Linux installation packages.

There's clearly a bug here though. Someone who can trigger it should try to fix it at least so we get an error-message that it didn't work. Because otherwise we make it harder to trigger it in the future.

**Comment by casella** [comment]
> There's clearly a bug here though (#9583). Someone who can trigger it should try to fix it at least so we get an error-message that it didn't work. Because otherwise we make it harder to trigger it in the future.

@sjoelund I agree with you in principle. Unfortunately I'm afraid that the set of persons that can trigger this issue and fix it is pretty much empty. In which case we'd be stalling the release of 1.20.0 indefinitely. This clearly doesn't work.

In any case, it is easy to reproduce the issue even after we add the cached libraries, one just needs to remove them manually from the installation directory 😅

**Comment by ramanunni** [comment]
Thanks a lot for the explanation, that goes a long way in clarifying things for me..

Currently, I have worked around this for Ubuntu (22.04) in docker, by just adding a couple of extra lines in the Dockerfile after the `apt-get install omc `(I was using 1.19.3). I ran a .mos (with installpackage for MSL) like shown in the code below (.mos created in the Dockerfile or added from a 'src' folder) to install the MSL libraries and just pointed USER's MODELICAPATH to the .openmodelica folder. I only added the MODELICAPATH variable to cover my bases as I was trying out Buildings 9.0.0 as well. It all seems to work perfectly fine now. Works with OMPython calls and runs files using Energyplus as well.

```
RUN omc $HOME/Mod/installscript.mos
RUN echo "export MODELICAPATH=${MODELICAPATH}:~/.openmodelica/libraries" >> ~/.bashrc
```
I felt this was the easiest way around for a minimal install. Please do correct me if I am wrong and there is an easier way around this. I hope I am not needlessly complicating things by doing this. 

I will test this for 1.18 as well over the next week or so. In addition, I also plan to test this for a Manjaro source build and report back.

**Comment by casella** [comment]
Tweaking the MODELICAPATH is certainly one option in 1.18.x and 1.19.x. When you start using 1.20.0 you can forget about all that 😃

**Comment by sjoelund** [comment]
> Tweaking the MODELICAPATH is certainly one option in 1.18.x and 1.19.x. When you start using 1.20.0 you can forget about all that 😃

Tweaking your MODELICAPATH should always be an option. The package manager is disabled if you do so

**Comment by casella** [comment]
Of course it is, but most likely you wont' need that 😅

**Comment by sjoelund** [comment]
Well, for docker there is a good point to do it. We should add some convenient pattern of creating your own images with fixed Modelica libraries of your choice. Something like `omc --tool=SystemPackageManager "Modelica 3.2.3"` which would install to `/usr/lib/omlibrary` and you could remove the cache files to have a small docker image.

**Comment by anargam** [comment]
> > Tweaking the MODELICAPATH is certainly one option in 1.18.x and 1.19.x. When you start using 1.20.0 you can forget about all that 😃
> 
> Tweaking your MODELICAPATH should always be an option. The package manager is disabled if you do so

@sjoelund Does this 'disabling' manifest as a `MODELICAPATH` issue (`$HOME/.openmodelica/libraries` not being on MODELICAPATH)? (This is based trying to construct a multi-directory MODELICAPATH under v1.23.1 under Linux)

**Comment by anargam** [comment]
> Tweaking the MODELICAPATH is certainly one option in 1.18.x and 1.19.x. When you start using 1.20.0 you can forget about all that 😃

This, if correct, is not entirely obvious from [the getModelicaPath() documentation](https://build.openmodelica.org/Documentation/OpenModelica.Scripting.getModelicaPath.html), and some other similar pages. They reference MODELICAPATH being
* a variable defaulting to a specific `$OPENMODELICAHOME`-based value. It  does not appear to be affected by the value of the OPENMODELICAHOME env variable. (v1.23.1)
* a colon-separated list. Setting MODELICAPATH to include a path that appears to work _and also_ either `$HOME/.openmodelica/libraries` or even an extra copy of the path, appears to breaks things. (That this was set via the [OPENMODELICALIBRARY](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/faq.html#openmodelica-general) variable was also somewhat surprising).

I may be mistaken in my rather brief and testing testing (not an OpenModelica user, in a hurry to package things for courses), so am probably missing things. But if the message above is correct, these particular parts of the docs may need some attention.

**Comment by casella** [comment]
getModelicaPath() only gives you information. I guess maybe we should be more explicit in the setModelicaPath() documentation about the consequences of using it. @anargam is that what you mean?

**Comment by anargam** [comment]
The page I linked to above appears to contain information about the variables themselves. Namely

> 
> The MODELICAPATH is a list of paths to search when trying to [load a library](modelica://OpenModelica.Scripting.loadModel). It is a string separated by colon (:) on all OSes except Windows, which uses semicolon (;).
> 
> To override the default path ([OPENMODELICAHOME](modelica://OpenModelica.Scripting.getInstallationDirectoryPath)/lib/omlibrary/:~/.openmodelica/libraries/), set the environment variable OPENMODELICALIBRARY=...

This information appears to be inconsistent with the description above.

Then there may be further consequences that I have no idea about. :)
